src/main/java/io/cryostat/recordings/RecordingHelper.java

```diff
@@ -117,6 +146,52 @@ public IRecordingDescriptor startRecording(
                 });
     }
 
+    public String saveRecording(ConnectionDescriptor connectionDescriptor, String recordingName)
+            throws Exception {
+
+        String saveName =
+                targetConnectionManager.executeConnectedTask(
+                        connectionDescriptor,
+                        connection -> {
+                            Optional<IRecordingDescriptor> descriptor =
+                                    getDescriptorByName(connection, recordingName);
+
+                            if (descriptor.isPresent()) {
+                                return writeRecordingToDestination(connection, descriptor.get());
+                            } else {
+                                throw new HttpStatusException(
```

Comment:
This is more of a style thing relating to loose coupling or encapsulation, but I think the `RecordingHelper` should avoid throwing `HttpStatusException`s. Here, for example, this should be something like a `RecordingNotFoundException`. The `TargetRecordingPatchSaveHandler` should then catch those and rethrow them as `HttpStatusException 404`.

Reply:
How should  `PeriodicArchiver.run()` handle `RecordingNotFoundException` from the helper functions and `Exception` from `targetConnectionManager.execute`? 
`run()` throws an `HttpStatusException` but I realized the archiver isn't part of the HTTP API. Should I use the `failureNotifier`?

Reply:
Yea, I don't think `run` should throw `HttpStatusException` either. But being that it's there because `PeriodicArchiver implements Runnable`, `run` can't be declared to throw any checked Exception type either. Any exceptions that occur need to be handled some other way than by (re)throwing. The purpose of the `failureNotifier` is so that the `RuleProcessor` can cancel the scheduled future runs of `PeriodicArchiver`s if they encounter failures, so that they don't continually retry and fail indefinitely. `RecordingNotFoundException` sounds like a good reason to stop running. The general `Exception` from `targetConnectionManager.execute` is harder to say - it seems like it might depend on what the exact problem is/was. Most likely, failures from `execute` will be because the target credentials were rejected, or the network connection could not be opened. If the credentials are rejected then it seems like stopping makes sense. If the network connection fails then it might make sense to just skip the current execution but not cancel altogether, because network problems are often transitory.

Reply:
> it seems like it might depend on what the exact problem is/was

How can I determine what caused `Exception`?

Reply:
https://commons.apache.org/proper/commons-lang/apidocs/org/apache/commons/lang3/exception/ExceptionUtils.html#hasCause-java.lang.Throwable-java.lang.Class-

You could do something like `if (ExceptionUtils.hasCause(ex, ConnectionException.class))`, for example. `AbstractAuthenticatedRequestHandler` does a few similar checks so that API handlers have consistent behaviour in various generalized failure cases like JMX connection errors, you could reference that for some inspiration.

Reply:
What do you think about only applying the `failureNotifier` for `RecordingNotFound` and credentials-related exceptions? Other `Exception`s will get logged and the scheduler will retry.

```
} catch (InterruptedException | ExecutionException | RecordingNotFoundException e) {

            logger.error(e);
            failureNotifier.apply(Pair.of(serviceRef, rule));
        } catch (Exception e) {
            logger.error(e);

            if (ExceptionUtils.hasCause(e, SecurityException.class)
                    || ExceptionUtils.hasCause(e, SaslException.class)) {

                failureNotifier.apply(Pair.of(serviceRef, rule));
            }
        }
```

Reply:
Seems to make sense to me.

Reply:
Thanks! PR is ready to go